### ISSUE TITLE
Analyzers don't handle out of order named parameters

### ISSUE THREAD

**Issue by tpodolak** [nsubstitute/NSubstitute.Analyzers#153]
All analyzers dont handle correctly out of order named parameters. For instance, for following piece of code
```c#
namespace TestProject6
{
    public class FooTests
    {
        public interface IFoo
        {
            int Foo();
        }

        public void Foo()
        {
            var substitute = Substitute.For<IFoo>();

            SubstituteExtensions.Returns(returnThis: 1, value: substitute.Foo());
        }
    }
}
```
NS1000 warning is reported as analyzer "think" that we want to setup `1` to return `substitute`
```
 SomeTest.cs(17, 54): [NS1000] Member 1 can not be intercepted. Only interface members and virtual, overriding, and abstract members can be intercepted.
```
In order to fix this, it is necessary to start inferring substituted member with either Operation API or SemanticModel
